gobblin-data-management/src/main/java/gobblin/data/management/retention/action/MultiAccessControlAction.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.data.management.retention.action;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.hadoop.fs.FileSystem;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigRenderOptions;
+
+import gobblin.data.management.retention.dataset.ConfigurableCleanableDataset;
+import gobblin.data.management.version.DatasetVersion;
+import gobblin.util.ConfigUtils;
+
+
+/**
+ * A wrapper around {@link AccessControlAction} that delegates the {@link #execute(List)} call to multiple embedded
+ * {@link AccessControlAction}s
+ * Use {@link MultiAccessControlActionFactory} to create new {@link MultiAccessControlAction}s
+ */
+public class MultiAccessControlAction extends RetentionAction {
+
+  private final List<AccessControlAction> embeddedAccessControlActions;
+  private static final String POLICIES_KEY = "policies";
+
+  /**
+   * The expected paths in the <code>actionConfig</code> are shown below.
+   * <pre>
+   * {
+   *      ## list all the policies, each policy should have a path in the config
+   *      policies = [restricted, .....]
+   *
+   *       restricted {
+   *           selection {
+   *             policy.class=gobblin.data.management.policy.SelectBeforeTimeBasedPolicy
+   *            timeBased.lookbackTime = 7d
+   *         }
+   *          mode : 750
+   *          owner : onr
+   *          group : grp
+   *      }
+   *   }
+   * </pre>
+   * @param actionConfig to use while creating a new {@link MultiAccessControlAction}
+   * @param fs
+   */
+  private MultiAccessControlAction(Config actionConfig, FileSystem fs) {
+    super(actionConfig, fs);
+    this.embeddedAccessControlActions = Lists.newArrayList();
+    for (String policy : ConfigUtils.getStringList(actionConfig, POLICIES_KEY)) {
+      Preconditions.checkArgument(actionConfig.hasPath(policy));
+      embeddedAccessControlActions.add(new AccessControlAction(actionConfig.getConfig(policy), fs));
```

Comment:
It may be worthwhile to check if the config is there and throw an exception with a meaningful message.